core/trino-spi/src/main/java/io/trino/spi/connector/SchemaTableName.java

```diff
@@ -75,4 +75,9 @@ public SchemaTablePrefix toSchemaTablePrefix()
     {
         return new SchemaTablePrefix(schemaName, tableName);
     }
+
+    public SchemaTableName asLegacySchemaTableName()
```

Comment:
This cannot got this way. `SchemaTableName` has to have clear semantic either it is lower cased or has a canonical form. It is too easy to forget about calling `asLegacySchemaTableName()`.